ionoscloud/resource_certificate_manager_provider.go

```diff
@@ -0,0 +1,199 @@
+package ionoscloud
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"slices"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/ionos-cloud/terraform-provider-ionoscloud/v6/services"
+	"github.com/ionos-cloud/terraform-provider-ionoscloud/v6/services/cert"
+	"github.com/ionos-cloud/terraform-provider-ionoscloud/v6/utils"
+	"github.com/ionos-cloud/terraform-provider-ionoscloud/v6/utils/constant"
+)
+
+func resourceCertificateManagerProvider() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: providerCreate,
+		ReadContext:   providerRead,
+		UpdateContext: providerUpdate,
+		DeleteContext: providerDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: providerImport,
+		},
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The name of the certificate provider",
+			},
+			"email": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The email address of the certificate requester",
+			},
+			"server": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The URL of the certificate provider",
+			},
+			"location": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The location of the certificate provider",
+			},
+			"external_account_binding": {
+				Type:     schema.TypeList,
+				MaxItems: 1,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"key_id": {
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "The key ID of the external account binding",
+						},
+						"key_secret": {
+							Type:        schema.TypeString,
+							Required:    true,
+							Sensitive:   true,
+							Description: "The secret of the external account binding",
+						},
+					},
+				},
+			},
+		},
+		CustomizeDiff: checkProviderImmutableFields,
+		Timeouts:      &resourceDefaultTimeouts,
+	}
+}
+
+func checkProviderImmutableFields(_ context.Context, diff *schema.ResourceDiff, _ interface{}) error {
```

Comment:
let's make these forcenew as is idiomatic. I know we have some older resources that throw errors, but new ones should force replace

Reply:
@cristiGuranIonos done in ec747230649ad96067a92e1d3b11d469d3eda2f1